src/modules/flow/http-server/http-server.c

```diff
@@ -994,6 +1032,164 @@ direction_vector_open(struct sol_flow_node *node, void *data,
+    struct sol_http_request *request, struct sol_http_response *response,
+    bool *updated)
+{
+    int r;
+    struct http_data *mdata = sol_flow_node_get_private_data(node);
+
+    if (sol_http_request_get_method(request) == SOL_HTTP_METHOD_POST) {
+        struct sol_blob *blob = NULL;
+        void *blob_mem = NULL;
+        struct sol_http_param_value *param;
+        uint16_t i;
+
+        SOL_HTTP_PARAMS_FOREACH_IDX (sol_http_request_get_params(request),
+            param, i) {
+            if (param->type != SOL_HTTP_PARAM_POST_DATA)
+                continue;
+
+            blob_mem = malloc(param->value.data.value.len);
+            SOL_NULL_CHECK(blob_mem, -ENOMEM);
+            memcpy(blob_mem, param->value.data.value.data,
+                param->value.data.value.len);
+            blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL,
+                blob_mem, param->value.data.value.len);
+            break;
+        }
+
+        if (!blob && blob_mem) {
+            free(blob_mem);
+            SOL_WRN("Could not create a blob to hold the json data");
+            return -ENOMEM;
+        } else if (!blob && !blob_mem) {
+            SOL_WRN("The POST request has no POST data!");
+            return -ENOENT;
+        }
+
+        r = replace_json_blob(mdata, blob);
+        sol_blob_unref(blob);
+        SOL_INT_CHECK(r, < 0, r);
+        *updated = true;
+    }
+
+    if (!mdata->value.blob) {
+        response->response_code = SOL_HTTP_STATUS_NOT_FOUND;
+        return 0;
+    }
+
+    r = sol_buffer_append_bytes(&response->content,
+        (uint8_t *)mdata->value.blob->mem, mdata->value.blob->size);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (!sol_http_param_add(&response->param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, HTTP_HEADER_CONTENT_TYPE_JSON))) {
+        SOL_WRN("Could not set the Content-Type:application/json");
+        return -ENOMEM;
+    }
+
+    return 0;
+}
+
+static void
+json_send_packet_cb(struct http_data *mdata,
+    struct sol_flow_node *node)
+{
+    struct sol_json_scanner object_scanner, array_scanner;
+
+    sol_json_scanner_init(&object_scanner, mdata->value.blob->mem,
+        mdata->value.blob->size);
+    sol_json_scanner_init(&array_scanner, mdata->value.blob->mem,
+        mdata->value.blob->size);
+
+    if (sol_json_is_valid_type(&object_scanner, SOL_JSON_TYPE_OBJECT_START)) {
+        sol_flow_send_json_object_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__OBJECT,
+            mdata->value.blob);
+    } else if (sol_json_is_valid_type(&array_scanner,
+        SOL_JSON_TYPE_ARRAY_START)) {
+        sol_flow_send_json_array_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__ARRAY,
+            mdata->value.blob);
+    } else {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Invalid json format - It will not be sent");
+        SOL_WRN("Invalid json format - It will not be sent");
```

Comment:
i guess the warning message is not necessary. If the error packet is not handled its message is printed.

In this case the message will be printed twice